parla/utils.py

```diff
@@ -0,0 +1,37 @@
+"""
+Generic, standalone utilities.
+
+All the stuff here should NOT depend on any other Parla-specific types/functions/etc.
+"""
+from typing import TypeVar, Tuple, Union, Iterable, Callable
+
+__all__ = ["traverse",]
+
+
+# TODO (bozhi): another parla module for typing?
+T = TypeVar('T') # (PEP 484) The argument to TypeVar() must be a string equal to the variable name to which it is assigned.
+
+
+def traverse(prefix: T, index: Tuple[Union[slice, Iterable[int], int]], step: Callable[[T, int], T], stop: Callable[[T], None]):
```

Comment:
`traverse` is not specific enough of a name. This is a pretty specific operation related to in nd indicies. The function name should reflect that.

Reply:
It's a legacy name and I noted this as a todo in the docstring below (line 24). I will consider a better name when updating the docs.